libevm/VMSIMD.cpp

```diff
@@ -0,0 +1,669 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <libethereum/ExtVM.h>
+#include "VMConfig.h"
+#include "VM.h"
+#if EIP_616
+
+namespace dev
+{
+namespace eth
+{
+
+// conversion functions to overlay vectors on storage for u256 stack slots
+// a dirty trick but it keeps the SIMD types from polluting the rest of the VM
+// so at least assert there's room for the trick, and use typedefs for some safety
+static_assert(sizeof(uint64_t[4]) <= sizeof(u256), "stack slot too narrow for SIMD");
+typedef uint64_t a64x4 [4];		
+typedef uint32_t a32x8 [8];
+typedef uint16_t a16x16[16];
+typedef uint8_t  a8x32 [32];
+      a64x4&  v64x4 (      u256& _stack_item) { return (a64x4&) *(a64x4*) &_stack_item; }
```

Comment:
I'm pretty sure `return reinterpret_cast<a64x4&>(_stack_item);` will do the same.